cluster-autoscaler/simulator/drainability/status.go

```diff
@@ -48,6 +48,23 @@ type Status struct {
 	BlockingReason drain.BlockingPodReason
 	// Error contains an optional error message.
 	Error error
+	// Interrupted means that the Rule returning the status exited early and that
```

Comment:
No it doesn't. It only means that remaining rules were not executed.

This is a result of the suggestion you made to factor out short-circuit validation checks (e.g. long-terminating, safe-to-evict, etc). These pods are indeed UndefinedOutcome, but they need to signal to the framework to short circuit (see line 91 in rules.go). Since the current API for drainability rules Drainable() function is to return a Status, it seems appropriate to add this intent to the Status object.

Reply:
It was necessary to deal with the branched logic, but it's no longer needed with the below refactor.